src/api/include/migraphx/migraphx.hpp

```diff
@@ -1255,7 +1273,9 @@ struct experimental_custom_op_base
     virtual std::string name() const                                            = 0;
     virtual argument compute(context ctx, shape output, arguments inputs) const = 0;
     virtual shape compute_shape(shapes inputs) const                            = 0;
-    virtual ~experimental_custom_op_base()                                      = default;
+    virtual std::ptrdiff_t output_alias(shapes) const { return -1; }
```

Comment:
`output_alias` should return a vector. We currently dont do this for our internal operators but we need to in order to support multiple aliases from tuples. Since we cant easily change APIs in the future, we should aim for the correct signature from the beginning.

Reply:
Added opaque object for `vector<ptrdiff>` similar to shapes, arguments, and instructions. 

I tried inferring the return type of virtual method but that didn't seem possible easily, without having type information of the input arguments for that method.  Idea was to switch between different methods for setting function pointers if the return type of vector.